airflow/www/api/experimental/endpoints.py

```diff
@@ -172,6 +175,85 @@ def task_info(dag_id, task_id):
     return jsonify(fields)
 
 
+@api_experimental.route('/dags/<string:dag_id>/tasks', methods=['GET'])
+@requires_authentication
+def get_dag_tasks(dag_id):
+    """Returns a JSON array of task ids of a given `dag_id`. """
+    try:
+        info = get_tasks(dag_id)
+    except AirflowException as err:
+        _log.info(err)
+        response = jsonify(error="{}".format(err))
+        response.status_code = err.status_code
+        return response
+
+    return jsonify(info)
+
+
+@api_experimental.route('/dags', methods=['GET'])
+@requires_authentication
+def dags():
```

Comment:
Would it be possible to add an option for some sort of filtering here, for example only returning DAGs with a provided prefix or matching a provided RegEx? It's fully possible that people might have hundreds to thousands of DAGs in which case they may not want to return the whole contents of the DagBag.

Reply:
Yes, I think that's a good idea. I will add these to the following methods as well.
GET - Tasks in a DAG
GET - Upstream Tasks
GET - Downstream Tasks